### ISSUE TITLE
Test failure: System.Net.Sockets.Tests.TimeoutTest / ReceiveTimesOut_Throws

### ISSUE THREAD

**Issue by davidsh** [dotnet/corefx#23767]
## Types of failures

Similar to #24804 (`SendTimesOut_Throws`)

Build 20170814.02:
```
Assert.InRange() Failure
Range:  (1500 - 3000)
Actual: 1346.6928

at System.Net.Sockets.Tests.TimeoutTest.ReceiveTimesOut_Throws(Boolean forceNonBlocking) in /root/corefx-923915/src/System.Net.Sockets/tests/FunctionalTests/TimeoutTest.cs:line 83
```

## History of failures

Day | Build | OS
-- | -- | --
8/14 | 20170814.02 | Debian90

# Original report

Test failed during PR run:
 ReceiveTimesOut_Throws(forceNonBlocking: True)
https://mc.dot.net/#/user/davidsh/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/169daaad2062ab623fdafa738c3fdbc778d3652e

ReceiveTimesOut_Throws(forceNonBlocking: True) (Ubuntu)
```
Assert.InRange() Failure
Range:  (1500 - 3000)
Actual: -15521.3745
```

**Comment by wfurt** [comment]
we can make the range more liberal like we did in #24844 for sending. 
As far as the negative value: Only one way how that can happen AFAIK is if time is changed during test run. It may be better to use something like Environment.TickCount than actual DateTime.Now.
We can also exclude this test from parallelism to get more predictable time behavior. 

I did not find any recent failure in related SendTimesOut_Throws. So there should be way how to make this test stable as well.
